### PR TITLE
deregister listeners when scope is destroyed

### PR DESCRIPTION
**Pull request by teaualune**

### CODE REVIEW

**Comment by kasperp** [comment]
@teaualune thanks for you pull request. Could you perhaps provide some information about why you need this functionality. What is it that you want to cleanup in the returned deregister function ?

**Comment by teaualune** [comment]
Hi @kasperp , thanks for your reply!
Actually I was working on another desired feature (implementation is done but neither documentation nor tests are prepared), so I made this pull request first, which improves performance and memory efficiency for ngReact.

**Comment by kasperp** [comment]
@teaualune sorry for the long wait, I kind of forgot about this to be honest. 

Anyway is this still relevant? Also could you perhaps enlighten me on how it improves performance ?

**Comment by teaualune** [comment]
@kasperp originally I thought that the watchers should be deregistered when scope is destroyed, since the watchers become redundant thereafter.

However after some survey (especially onto [this post](http://stackoverflow.com/questions/25113884/should-angular-watch-be-removed-when-scope-destroyed)), it seems that the `$watchCollection` listeners will be auto deregistered after the scope is destroyed (but I'm not pretty sure about this here).

To sum up, I guess this PR does some redundant things and can be ignored :sweat_smile:

**Comment by kasperp** [comment]
thanks for getting back :smile:
